src/hotspot/share/compiler/compilationPolicy.hpp

```diff
@@ -34,12 +34,18 @@ class CompileTask;
 class CompileQueue;
 /*
  *  The system supports 5 execution levels:
- *  * level 0 - interpreter
+ *  * level 0 - interpreter (Profiling information stored in a MethodData object, or MDO)
  *  * level 1 - C1 with full optimization (no profiling)
  *  * level 2 - C1 with invocation and backedge counters
- *  * level 3 - C1 with full profiling (level 2 + MDO)
+ *  * level 3 - C1 with full profiling (level 2 + All data available from an MDO)
  *  * level 4 - C2
  *
+ * Note: The MethodData object is specific to the interpreter, when execution progresses to
+ * levels 2 and 3, profile data collection utilizes the separate ciMethodData instead. If any
+ * profiling information was collected from the interpreter (The method has a non-null MDO),
+ * all data from the MDO will be loaded into the ciMethodData when it is first created.
+ * (See ciMethod::method_data() in ciMethod.cpp for more details)
+ *
```

Comment:
The ciMethodData is just a temporary snapshot.  Updates to the profiling data is still done through the MethodData.

Reply:
The only place I could find a MethodData being created is in the method
```
// Build a MethodData* object to hold information about this method
// collected in the interpreter.
void Method::build_interpreter_method_data(const methodHandle& method, TRAPS)
```
which presumably constructs a MethodData for profiling in the Interpreter. Is there a different area where it's created (when profiling with C1) that I missed?

Reply:
hopefully resolved by the newest changes